design/move-plugin-repos.md

```diff
@@ -0,0 +1,110 @@
+
+## Action items
+
+### Todo list
+
+#### Repository creation
+
+- [ ] Use GH UI to create each repository in the new VMW org. Who: new org owner; TBD
+- [ ] Make owners of the Velero repo owners of each repo in the new org. Who: new org owner; TBD
+- [ ] Add Travis CI. Who: Any of the new repo owners; TBD
+- [ ] Add webhook: travis CI. Who: Any of the new repo owners; TBD
+- [ ] Add DCO for signoff check (https://probot.github.io/apps/dco/). Who: Any of the new repo owners; TBD
+
+#### Plugin changes
+
+- [ ] Modify Velero so it can install any of the provider plugins. https://github.com/heptio/velero/issues/1740 - Who: @nrb
+- [ ] Extract each provider plugin into their own repo. https://github.com/heptio/velero/issues/1537
+- [ ] Create deployment and grpc-push scripts with the new location path. Who: @carlisia
+- [ ] Add documentation for how to use the plugin. Who: @carlisia
+- [ ] Update Helm chart to install Velero using any of the provider plugin. https://github.com/heptio/velero/issues/1819
+
+### Notes/How-Tos
+
+#### Creating the GH repository
+
+[Pending] The organization owner will make all current owners in the Velero also repo owners in each of the new org plugin repos.
+
+#### Setting up Travis CI
+
+Someone with owner permission on the new repository needs to go to their Travis CI account and authorize Travis CI on the repo. Here are instructions: https://docs.travis-ci.com/user/tutorial/.
+
+After this, any webhook notifications can be added following these instructions: https://docs.travis-ci.com/user/notifications/#configuring-webhook-notifications.
+
+## High-Level Design
+
+Each provider plugin will be an independent project, using the Velero library to implement their specific functionalities.
+
+The way Velero is installed will be changed to accommodate installing these plugins at deploy time, namely the Velero `install` command, as well as the Helm chart.
+
+Each plugin repository will need to have their respective images built and pushed to the same registry as the Velero images.
+
+## Detailed Design
+
+### Projects
+
+Each provider plugin will be an independent GH repository, named: `velero-plugin-aws`, `velero-plugin-azure`, and `velero-plugin-gcp`.
+
+Build of the project will be done the same way as with Velero, using Travis.
+
+Images for all the plugins will be pushed to the same repository as the Velero image, also using Travis.
+
+Releases of new versions can occur as needed and independent of Velero releases. However, a matrix documenting which plugin versions are compatible with each Velero version should be added to the docs.
+
+Documentation for how to install and use the plugins will be augmented in the existing Plugins section of the Velero documentation.
```

Comment:
Another question - what will we do with documentation? We currently have setup docs for these providers on `velero.io`. Do we plan to move the docs into the plugin repos? If so, how do we ensure users can find them? Do we want to allow docs for other third-party plugins onto velero.io?

Reply:
This is a really good question. At the moment, I'd lean towards allowing documentation for all plugins onto velero.io, and we maintain these 3.

I'm not sure if we could include pages from other repos into a Netlify site, @jonasrosland may have a better idea.

Reply:
From our discussion last Friday, I think we agreed on simply linking from the velero docs to the plugin docs - is that accurate? If so, I think this can be resolved